### PR TITLE
Kernel upgrade

### LINKED ISSUE #185: New kprobes.
Kernel `5.10` removed old helper `do_fork` and brought the helper `kernel_clone`, I also observed that after kernel `5.5.19` it does not make sense to attach the kprobe to the function `do_sys_open`, instead we must add to the function `do_sys_openat2`, because our count is not accurate.

### LINKED ISSUE #186: Rename files
I observed that the eBPF programs that we are using should not have the patch version, because we are saying for users that we are loading `5.4.20` on `5.4.85`. 

It would be good to simplify the name removing the patch. In reality we only need to know the kernel version that we compiled the code.

### PR DESCRIPTION
**Pull request by thiagoftsm**
Fixes #185 
Fixes #186

The kernel team made important changes with the release of the latest kernel( 5.10.0), this PR brings the necessary changes for we keep showing the same charts on this new kernel.

I am also using this PR to upgrade the kernel versions used for other Long Term Support Kernel. (@Ferroin github is keeping the old kernels `5.4.20` and `4.14.171`, but I already removed them from files inside `.github`. What am I missing?)

This PR was tested with the following Linux distributions

| Linux Distribution | Kernel version |
|-------------------------|---------------------|
| Slackware current | 5.10.3 |
| Slackware current |  5.9.11|
| Slackware current |5.8.18 |
| Slackware current |5.7.19 |
| Slackware current |5.6.19 |
| Slackware current |4.20.17 |
| Ubuntu 20.04 | 5.4.0-58-generic |
| Ubuntu 18.04 | 5.4.0-58-generic |
| Ubuntu 18.04 |4.18.0-25-generic |
| Ubuntu 18.04 |4.15.0-128-generic|
|CentOS 7 | 3.10.0-1160.6.1.el7.x86_6|

Using the PR https://github.com/netdata/netdata/pull/10434 .

### CODE REVIEW

File: kernel/process_kern.c [hunk 1]
```diff
@@ -739,6 +749,79 @@ int netdata_clone(struct pt_regs* ctx)
...
+    struct netdata_pid_stat_t data = { };
+    struct netdata_pid_stat_t *fill;
+    __u64 pid_tgid = bpf_get_current_pid_tgid();
+    __u32 pid = (__u32)(pid_tgid >> 32);
+    __u32 tgid = (__u32)( 0x00000000FFFFFFFF & pid_tgid);
+
+    int threads = 0;
+    struct kernel_clone_args *args = (struct kernel_clone_args *)PT_REGS_PARM1(ctx); 
+    __u64 clone_flags;
+    bpf_probe_read(&clone_flags, sizeof(clone_flags), (void *)&clone_flags);
```
Comment (vlvkobal) [inline kernel/process_kern.c:774]: Did you want to use `args` as the source?

File: kernel/process_kern.c [hunk 2]
```diff
@@ -739,6 +749,79 @@ int netdata_clone(struct pt_regs* ctx)
...
+    if (clone_flags & CLONE_VM) {
```
Comment (vlvkobal) [inline kernel/process_kern.c:775]: What happened with the `CLONE_THREAD` flag? Isn't it used in the new kernel anymore?

File: kernel/process_kern.c (hunk 1 above)
Comment (thiagoftsm) [reply kernel/process_kern.c]: You are right I had wrong typo, I also changed the variable, instead to use the whole flags, I am using `exit_signal` that is present in the first 8 bits of the `args->flags`, because I observed a difference when I was using only flags when I compated the kernel `5.10.4` with the kernel `5.9.16`.

Comment (thiagoftsm) [reply kernel/process_kern.c]: Thank you!

File: kernel/process_kern.c (hunk 2 above)
Comment (thiagoftsm) [reply kernel/process_kern.c]: When I saw `kernel_thread`, I observed that `CLONE_THREAD` was not set inside the function, but after to take a look on other functions, I saw that it was not discarded. So I misunderstood the changes made by the kernel team in the latest version. The fix was done with the changes I brought with the last commit.

File: kernel/process_kern.c [hunk 3]
```diff
@@ -822,6 +758,72 @@ int netdata_sys_clone(struct pt_regs *ctx)
 
 #endif
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,3,0))
```
Comment (vlvkobal) [inline kernel/process_kern.c:761]: Why did you move this block of code? The new place is incorrect. Both sections will be defined - `kretprobe/kernel_clone` and `kretprobe/sys_clone`.

BTW, on my system `/usr/include/linux/version.h` contains an incorrect `LINUX_VERSION_CODE`. The correct one is in `/usr/lib/modules/5.10.4-arch2-1/build/include/generated/uapi/linux/version.h`.

Comment (thiagoftsm) [reply kernel/process_kern.c]: > Why did you move this block of code? The new place is incorrect. Both sections will be defined - 

I moved it last night to test and I forgot to return, thank you to call the attention. 

> BTW, on my system /usr/include/linux/version.h contains an incorrect LINUX_VERSION_CODE

On Slackware I tested now, and the correct version is set inside `/usr/include/linux/version.h`

```bash
bash-5.1$ cat /usr/include/linux/version.h
#define LINUX_VERSION_CODE 330244
#define KERNEL_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
bash-5.1$ bc
bc 1.07.1
Copyright 1991-1994, 1997, 1998, 2000, 2004, 2006, 2008, 2012-2017 Free Software Foundation, Inc.
This is free software with ABSOLUTELY NO WARRANTY.
For details type `warranty'. 
5*65536 + 10*256 + 4
330244
```

It is possible your distribution changed the file, but I do not think this is affecting us, because we are using Ubuntu on our docker images and we are also adding the correct [path](https://github.com/netdata/kernel-collector/blob/446b4f0a49c2e9454a4fb1217cb2c88803f88a81/kernel/Makefile#L16) during the compilation time.

**Review by vlvkobal — APPROVED** [review]
I haven't tested it, but the code looks good to me.

**Review by underhood — APPROVED** [review]
